src/utils/graphQLClient.tsx

```diff
@@ -33,12 +50,53 @@ const sessionId = uuid();
 
 export const client = new Client({
   url: CHROMATIC_API_URL,
-  exchanges: [fetchExchange], // no cacheExchange to prevent sharing data between stories
+  exchanges: [
+    // no cacheExchange to prevent sharing data between stories
+    mapExchange({
+      onResult(result) {
+        if (result.data?.viewer === null) setCurrentToken(undefined);
+      },
+    }),
+    authExchange(async (utils) => {
+      return {
+        addAuthToOperation(operation) {
+          if (!currentToken) return operation;
+          return utils.appendHeaders(operation, { Authorization: `Bearer ${currentToken}` });
+        },
+
+        // Determine if the current error is an authentication error.
+        didAuthError: (error) =>
+          error.response.status === 401 ||
+          error.graphQLErrors.some((e) => e.message.includes("Must login")),
+
+        // If didAuthError returns true, clear the token. Ideally we should refresh the token here.
+        // The operation will be retried automatically.
+        async refreshAuth() {
+          setCurrentToken(undefined);
+        },
+
+        // Prevent making a request if we know the token is missing, invalid or expired.
+        // This handler is called repeatedly so we avoid parsing the token each time.
+        willAuthError() {
+          if (!currentToken) return true;
+          try {
+            if (!currentTokenExpiration) {
+              const { exp } = JSON.parse(atob(currentToken.split(".")[1]));
+              currentTokenExpiration = exp;
+            }
```

Comment:
Isn't this bit redundant? Don't we always set it above? At the least we should share this code.

Reply:
Ah yes, I forgot to clean this up after I made sure `currentTokenExpiration` is always set.